### PR TITLE
Rex shared event

### PR DESCRIPTION
**Pull request by Eyap53**
## Enhancement : 
- When manager want to finish event, 3 possibilities : (#14)
 -- Calculate price by divising total
 -- Fix price by ponderation
 -- No payment (previous finish)

- Managers can know see the total number of registration / participation (#14)
- Users can be listed by years (#12)

## Change : 
- Fusion of ProceedPayment and Finish.

## Fix : 
- Ajax to get price per users in ManageUsers
- Removing all users
- Some permissions et verifications

### CODE REVIEW

**Comment by AlexandrePalo** [comment]
Review en cours! 👍

File: finances/models.py [hunk 1]
```diff
@@ -599,7 +599,7 @@ def pay(self, operator, recipient):
 
         # Calcul du prix par weight
         total_weight = self.get_total_weights_participants()
-        final_price_per_weight = round(self.price / total_weight, 2)
+        final_price_per_weight = round(total_price / total_weight, 2)
```
Comment (AlexandrePalo) [inline finances/models.py]: Check ZeroDivision here ?

File: finances/templates/finances/js/calculate_users_price.html [hunk 2]
```diff
@@ -13,7 +11,7 @@
   };
 
   function set_prices () {
-    var total_price = get_total_price();
+    var total_price = Number({{price}});
```
Comment (AlexandrePalo) [inline finances/templates/finances/js/calculate_users_price.html]: If the price can be Decimal, use `{{ price I unlocalize }}` (module l18n). Thus, `1,18` will always show `1.18` (it's not always the case due to localization), needed by JS.

File: finances/models.py (hunk 1 above)
Comment (Eyap53) [reply finances/models.py]: Well, currently total_weight can't be null (it is checked in dispatch view).
I added the check "just to be sure"

File: finances/templates/finances/js/calculate_users_price.html (hunk 2 above)
Comment (Eyap53) [reply finances/templates/finances/js/calculate_users_price.html]: Didn't know that, thanks !

File: finances/models.py (hunk 1 above)
Comment (Eyap53) [reply finances/models.py]: Fix: 91273568d48f65078cbc4a51ea84f20e78b18b1f

File: finances/templates/finances/js/calculate_users_price.html (hunk 2 above)
Comment (Eyap53) [reply finances/templates/finances/js/calculate_users_price.html]: Fix: 8559ae2

**Comment by Eyap53** [comment]
Bugs to Fix still remaining : 

- [ ] In ordering participants, when a user doesn't have the attribute. -> I don't have any error when i try, did you had one ? 

- [ ] Currently, in transaction history the price displayed for payment by ponderation

**Comment by AlexandrePalo** [comment]
>  In ordering participants, when a user doesn't have the attribute. -> I don't have any error when i try, did you had one ?

I don't have the error anymore hmmm.
So it's ok for me :)

**Pull request merged**